src/coins.cpp

```diff
@@ -443,6 +443,18 @@ double CCoinsViewCache::GetPriority(const CTransaction &tx, int nHeight) const
             dResult += coins->vout[txin.prevout.n].nValue * (nHeight-coins->nHeight);
         }
     }
+
+    // Joinsplits do not reveal any information about a note's amount or age.
+    // We need to ensure priority increases the longer a tx remains in the mempool, so we don't
+    // generate random values which would change on each invocation of this method.
+    // Instead, we use the anchor as a seed to generate the note value deterministically.
+    // Currently, we say that each JoinSplit has input notes of between 1 to 2 ZEC,
+    // and that the age of each note is half the age of the blockchain.
+    for (auto & jsdesc : tx.vjoinsplit) {
+        uint64_t i = jsdesc.anchor.GetCheapHash();
```

Comment:
Could have used the txid hash instead of the anchor.  Or the size in bytes of the tx.  Just wanted to add some variation to the note value, rather than setting the note value to 1 ZEC.
Maybe GetHash() is better than GetCheapHash() ?